commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MeanTest.java

```diff
@@ -0,0 +1,316 @@
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testMean")
+    void testMeanRandomOrder(double[] values) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        for (int i = 1; i <= 10; i++) {
+            testMean(TestHelper.shuffle(rng, values));
+            testParallelStream(TestHelper.shuffle(rng, values));
+        }
+    }
+
+    static Stream<Arguments> testMean() {
+        return Stream.of(
+            Arguments.of(new double[] {0.0}),
+            Arguments.of(new double[] {10, 8, 13, 9, 11, 14, 6, 4, 12, 7, 5}),
+            Arguments.of(new double[] {8.04, 6.95, 7.58, 8.81, 8.33, 9.96, 7.24, 4.26, 10.84, 4.82, 5.68}),
+            Arguments.of(new double[] {9.14, 8.14, 8.74, 8.77, 9.26, 8.10, 6.13, 3.10, 9.13, 7.26, 4.74, 7.46, 6.77, 12.74, 7.11, 7.81, 8.84, 6.08, 5.39, 8.15, 6.42, 5.73}),
+            Arguments.of(new double[] {8, 8, 8, 8, 8, 8, 8, 19, 8, 8, 8}),
+            Arguments.of(new double[] {6.58, 5.76, 7.71, 8.84, 8.47, 7.04, 5.25, 12.50, 5.56, 7.91, 6.89}),
+            Arguments.of(new double[] {0, 0, 0.0}),
+            Arguments.of(new double[] {1, -7, 6}),
+            Arguments.of(new double[] {1, 7, -15, 3}),
+            Arguments.of(new double[] {2, 2, 2, 2}),
+            Arguments.of(new double[] {2.3}),
+            Arguments.of(new double[] {3.14, 2.718, 1.414}),
+            Arguments.of(new double[] {12.5, 12.0, 11.8, 14.2, 14.9, 14.5, 21.0, 8.2, 10.3, 11.3, 14.1, 9.9, 12.2, 12.0, 12.1, 11.0, 19.8, 11.0, 10.0, 8.8, 9.0, 12.3}),
+            Arguments.of(new double[] {-0.0, +0.0}),
+            Arguments.of(new double[] {0.0, -0.0}),
+            Arguments.of(new double[] {0.0, +0.0}),
+            Arguments.of(new double[] {0.001, 0.0002, 0.00003, 10000.11, 0.000004}),
+            Arguments.of(new double[] {10E-50, 5E-100, 25E-200, 35.345E-50}),
+            // Overflow of the sum
+            Arguments.of(new double[] {Double.MAX_VALUE, Double.MAX_VALUE}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {Double.MAX_VALUE, 1}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}),
+            Arguments.of(new double[] {Double.MAX_VALUE, -1}),
+            Arguments.of(new double[] {Double.MAX_VALUE, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {1, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {1, 1, 1, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {Double.MAX_VALUE, Double.MAX_VALUE / 2})
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testMeanNonFinite")
+    void testMeanNonFinite(double[] values, double expected) {
+        Mean mean = Mean.create();
+        for (double value : values) {
+            mean.accept(value);
+        }
+        TestHelper.assertEquals(expected, mean.getAsDouble(), 0, () -> "mean non-finite");
```

Comment:
When explicitly testing non-finite values we should use `Assertions.assertEquals` without a delta value.